public/app/features/trails/RelatedLogs/RelatedLogsScene.tsx

```diff
@@ -48,14 +51,21 @@ export class RelatedLogsScene extends SceneObjectBase<RelatedLogsSceneState> {
           }),
         ],
       }),
-      connectors: [lokiRecordingRulesConnector],
+      connectors: [],
       ...state,
     });
 
     this.addActivationHandler(this.onActivate.bind(this));
   }
 
   private onActivate() {
+    this.setState({
+      connectors: [lokiRecordingRulesConnector, createLabelsCrossReferenceConnector(this)],
+    });
+    this.setLogsDataSourceVar();
+  }
```

Comment:
Super cool! I see we iterate through the connectors, or the ways that we relate logs. This makes sense. So if I have 10 loki data sources, I will see 10 queries to each loki instance for rules and ten queries again for the filters as loki queries.

I think this is ok since it is still behind a feature toggle but I wonder about finding the largest amount of loki instances a customer might have. I have a feeling that this should be ok though, because a. I can't think of a reason customers would have 100s of lokis and b. even in the above scenario with 20 queries total, that is similar the metric preview scene where we do tens on queries to prometheus.

Reply:
Great point! In https://github.com/grafana/grafana/pull/98775/commits/cabe271bfad917111382a96e9ee0ce30568a5e7f, I limited the number of Loki data sources to an arbitrary value of `10`. Let's see how that goes...we can raise it if that's what customers want.